src/data/enum.lean

```diff
@@ -0,0 +1,190 @@
+
+import category.monad.basic
+import data.list.basic
+import data.equiv.basic
+import data.finset
+import data.fintype
+
+open finset (hiding singleton)
+
+namespace fin
+
+def enum (n : ℕ) : list (fin n) :=
+(list.range n).attach.map $ λ ⟨i,h⟩, ⟨i,by simpa using h⟩
```

Comment:
Also `list.of_fn id`.

Reply:
I like the idea but `mem_enum` is easier to prove with my definition.